src/opm/output/eclipse/AggregateAquiferData.cpp

```diff
@@ -0,0 +1,510 @@
+
+#include <opm/parser/eclipse/EclipseState/Schedule/SummaryState.hpp>
+
+#include <opm/parser/eclipse/EclipseState/Tables/FlatTable.hpp>
+
+#include <opm/parser/eclipse/Units/UnitSystem.hpp>
+
+#include <opm/common/OpmLog/OpmLog.hpp>
+
+#include <fmt/format.h>
+
+#include <array>
+#include <cstddef>
+#include <numeric>
+#include <stdexcept>
+#include <vector>
+
+namespace VI = Opm::RestartIO::Helpers::VectorItems;
+
+namespace {
+    template <typename AquiferCallBack>
+    void CarterTracyAquiferLoop(const Opm::AquiferConfig& aqConfig,
+                                AquiferCallBack&&         aquiferOp)
+    {
+        for (const auto& aqData : aqConfig.ct()) {
+            aquiferOp(aqData);
+        }
+    }
+
+    template <typename AquiferCallBack>
+    void FetkovichAquiferLoop(const Opm::AquiferConfig& aqConfig,
+                              AquiferCallBack&&         aquiferOp)
+    {
+        for (const auto& aqData : aqConfig.fetp()) {
+            aquiferOp(aqData);
+        }
+    }
+
+    template <typename ConnectionCallBack>
+    void analyticAquiferConnectionLoop(const Opm::AquiferConfig& aqConfig,
+                                       ConnectionCallBack&&      connectionOp)
+    {
+        for (const auto& [aquiferID, connections] : aqConfig.connections().data()) {
+            const auto tot_influx =
+                std::accumulate(connections.begin(), connections.end(), 0.0,
+                    [](const double t, const Opm::Aquancon::AquancCell& connection) -> double
+                {
+                    return t + connection.influx_coeff;
+                });
+
+            auto connectionID = std::size_t{0};
+
+            for (const auto& connection : connections) {
+                connectionOp(aquiferID, connectionID++, tot_influx, connection);
+            }
+        }
+    }
+
+    namespace IntegerAnalyticAquifer
+    {
+        Opm::RestartIO::Helpers::WindowedArray<int>
+        allocate(const Opm::RestartIO::InteHEAD::AquiferDims& aqDims)
+        {
+            using WA = Opm::RestartIO::Helpers::WindowedArray<int>;
+
+            return WA {
+                WA::NumWindows{ static_cast<WA::Idx>(aqDims.maxAquiferID) },
+                WA::WindowSize{ static_cast<WA::Idx>(aqDims.numIntAquiferElem) }
+            };
+        }
+
+        namespace CarterTracy
+        {
+            template <typename IAaqArray>
+            void staticContrib(const Opm::AquiferCT::AQUCT_data& aquifer,
+                               const int                         numActiveConn,
+                               IAaqArray&                        iaaq)
+            {
+                using Ix = VI::IAnalyticAquifer::index;
+
+                iaaq[Ix::NumAquiferConn] = numActiveConn;
+                iaaq[Ix::WatPropTable] = aquifer.pvttableID; // One-based (=AQUCT(10))
+
+                iaaq[Ix::CTInfluenceFunction] = aquifer.inftableID;
+                iaaq[Ix::TypeRelated1] = 1; // =1 for Carter-Tracy
```

Comment:
Better give it a name - `AquiferUnknown1` - than using a magic number.

Reply:
I've pushed an update to address this.